utils/include/detray/simulation/event_generator/uniform_track_generator.hpp

```diff
@@ -63,6 +63,8 @@ class uniform_track_generator
         /// Magnitude of momentum: Default is one to keep directions normalized
         /// if no momentum information is needed (e.g. for a ray)
         scalar m_p_mag{1.f * unit<scalar>::GeV};
+        /// Whether to interpret the momentum @c m_p_mag as p_T
+        bool m_is_pT{true};
```

Comment:
Could you explain why we need this option?
Is it possible we just have separate options for p and pT range.

Reply:
All in all, I would not change the default behaviour of the generator which will interfere with traccc as well :thinking:

Reply:
I though having the option to generate tracks with constant p_T might be helpful to reason about curvature in xy, which I think is the main reason of failure in the navigation, but maybe it is not needed. I will revert the default

Reply:
> I though having the option to generate tracks with constant p_T

I see. Please do it with by fixing p and eta.

Reply:
OK, done. I changed as little as possible in the interface, so that things in traccc stay the same